Sources/Host/POSIX/PTrace.cpp

```diff
@@ -74,6 +74,54 @@ ErrorCode PTrace::detach(ProcessId pid) {
   return kSuccess;
 }
 
+ErrorCode PTrace::step(ProcessThreadId const &ptid, ProcessInfo const &pinfo,
+                       int signal, Address const &address) {
+  pid_t pid;
+
+  ErrorCode error = ptidToPid(ptid, pid);
+  if (error != kSuccess)
+    return error;
+
+#if defined(OS_LINUX)
```

Comment:
I don't like this at all. =[

What's the point of all this subclass stuff anyway when only one implementation is actually compiled? @_@